pyccel/codegen/printing/ccode.py

```diff
@@ -785,71 +776,89 @@ def _print_MathFunctionBase(self, expr):
+            self._additional_imports.add('pyc_math')
+        else:
+            if expr.dtype is NativeComplex():
+                self._additional_imports.add('cmath')
+            else:
+                self._additional_imports.add('math')
         args = []
         for arg in expr.args:
-            if arg.dtype is NativeComplex():
-                self._additional_imports.add('complex')
-            if arg.dtype is not NativeReal():
-                args.append(self._print(NumpyFloat(arg)))
-            else :
+            if arg.dtype != expr.dtype:
+                cast_func = python_builtin_datatypes[str_dtype(expr.dtype)]
+                args.append(self._print(cast_func(arg)))
+            else:
                 args.append(self._print(arg))
         code_args = ', '.join(args)
         return '{0}({1})'.format(func_name, code_args)
 
-    def _print_MathSqrt(self, expr):
+    def _print_MathCeil(self, expr):
+        """Convert a Python expression with a math ceil function call to C
+        function call"""
         # add necessary include
         self._additional_imports.add('math')
         arg = expr.args[0]
-        if arg.dtype is not NativeReal():
-            code_args = self._print(NumpyFloat(arg))
-        else :
-            code_args = self._print(arg)
-        return 'sqrt({})'.format(code_args)
-
-    def _print_PycMathFunctionBase(self, expr):
-        """ Convert a Python expression with a Pyc_math functions call to C
-        functions call in pyc_math library
+        if arg.dtype is NativeInteger():
+            code_arg = self._print(PythonFloat(arg))
+        else:
+            code_arg = self._print(arg)
+        return "ceil({})".format(code_arg)
 
-        Parameters
-        ----------
-            expr : Pyccel ast node
-                Python expression with a Math function call
+    def _print_MathIsfinite(self, expr):
+        """Convert a Python expression with a math isfinite function call to C
+        function call"""
+        # add necessary include
+        self._additional_imports.add('math')
+        arg = expr.args[0]
+        if arg.dtype is NativeInteger():
+            code_arg = self._print(PythonFloat(arg))
+        else:
+            code_arg = self._print(arg)
+        return "isfinite({})".format(code_arg)
 
-        Returns
-        -------
-            string
-                Equivalent expression in C language
+    def _print_MathIsinf(self, expr):
+        """Convert a Python expression with a math isinf function call to C
+        function call"""
+        # add necessary include
+        self._additional_imports.add('math')
+        arg = expr.args[0]
+        if arg.dtype is NativeInteger():
+            code_arg = self._print(PythonFloat(arg))
+        else:
+            code_arg = self._print(arg)
+        return "isinf({})".format(code_arg)
 
-        ------
-        Example:
-        --------
-            math.gcd(x, y) ==> pyc_gcd(x, y)
+    def _print_MathIsnan(self, expr):
```

Comment:
I'm sure some of these Math functions could be combined with a well chosen superclass (e.g. MathIsVal) but I think we can leave this for now